src/d3-ext/profile.js

```diff
@@ -0,0 +1,320 @@
+        xFactor = 1;
+        units = 'm';
+      }
+
+      xAxis.tickFormat(function(d) {
+        return d / xFactor;
+      });
+
+      g.select('.x.axis')
+          .transition()
+          .call(xAxis);
+
+      g.select('.x.label')
+          .text('distance (' + units + ')')
+          .style('fill', 'grey')
+          .style('shape-rendering', 'crispEdges');
+
+      g.select('.y.axis')
+          .transition()
+          .call(yAxis);
+
+      g.select('.grid-y')
+          .transition()
+          .call(yAxis.tickSize(-width, 0, 0).tickFormat(''))
+          .selectAll('.tick line')
+          .style('stroke', '#ccc')
+          .style('opacity', 0.7);
+
+      g.selectAll('.axis').selectAll('path, line')
+          .style('fill', 'none')
+          .style('stroke', '#000')
+          .style('shape-rendering', 'crispEdges');
+
+      g.selectAll('.grid-hover line')
+          .style('stroke', '#222')
+          .style('opacity', 0.8);
+
+      g.select('.overlay')
+          .on('mouseout', mouseout)
+          .on('mousemove', mousemove);
+
+      function mousemove() {
+        var mouseX = d3.mouse(this)[0];
+        var x0 = x.invert(mouseX);
+        var i = bisectDistance(data, x0, 1);
+
+        var point = data[i];
+        var elevation = point['values']['mnt'];
+        var dist = point['dist'];
+
+        g.select('.x.grid-hover')
+            .style('display', 'inline')
+            .select('line')
+            .attr('x1', x(dist))
+            .attr('y1', height)
+            .attr('x2', x(dist))
+            .attr('y2', y(elevation));
+
+        g.select('.y.grid-hover')
+            .style('display', 'inline')
+            .select('line')
+            .attr('x1', x(0))
+            .attr('y1', y(elevation))
+            .attr('x2', width)
+            .attr('y2', y(elevation));
+
+        var right = x0 > xDomain[1] / 2;
+        var xtranslate = x(dist);
+        xtranslate += right ? -10 : 10;
+
+        g.select('.x.grid-hover text')
+            .text(parseFloat(dist.toPrecision(3) / xFactor) + ' ' + units)
+            .style('text-anchor', right ? 'end' : 'start')
+            .attr('transform', 'translate(' + xtranslate + ',' +
+                (height - 10) + ')');
+
+        g.select('.y.grid-hover text')
+            .text(Math.round(elevation) + ' m')
+            .style('text-anchor', right ? 'end' : 'start')
+            .attr('transform', 'translate(' + xtranslate + ',' +
+                (y(elevation) - 10) + ')');
+        hover.call(null, point);
+      }
+
+      function mouseout() {
+        g.selectAll('.grid-hover')
+            .style('display', 'none');
+        if (out) {
+          out.call(null);
+        }
+      }
+    });
+  };
+
+  profile.hover = function(cb) {
+    if (!arguments.length) {
+      return hover;
+    }
+    hover = cb;
+    return profile;
+  };
+
+  profile.out = function(cb) {
+    if (!arguments.length) {
+      return out;
+    }
+    out = cb;
+    return profile;
```

Comment:
profile != this

`profile` is a function here
`this` is the ngeo.profile instance

It can be changed of course.